src/katsdpk8spoc/controller_server.py

```diff
@@ -0,0 +1,223 @@
+    responses:
+        "200":
+            $ref: '#/components/responses/Reply200Ack'
+        "405":
+            $ref: '#/components/responses/HTTPMethodNotAllowed'
+        "421":
+            $ref: '#/components/responses/HTTPMisdirectedRequest'
+        "422":
+            $ref: '#/components/responses/HTTPUnprocessableEntity'
+    """
+    response = await start_subarray(
+        request.query["subarray"], request.query["receptors"]
+    )
+    buf = html_page("start", request.query["subarray"], data=response)
+    return web.Response(body=buf, content_type="text/html")
+
+
+async def stop_handle(request):
+    """
+    ---
+    description: stop a subarray
+    parameters:
+       - in: query
+         name: subarray
+         schema:
+           type: integer
+         required: true
+         description: The subarray ID.
+    responses:
+        "200":
+            $ref: '#/components/responses/Reply200Ack'
+        "405":
+            $ref: '#/components/responses/HTTPMethodNotAllowed'
+        "421":
+            $ref: '#/components/responses/HTTPMisdirectedRequest'
+        "422":
+            $ref: '#/components/responses/HTTPUnprocessableEntity'
+    """
+    status = await subarray_status(request.query["subarray"])
+    items = status.get("items")
+    info = []
+    if items:
+        for wf in items:
+            res = await stop_workflow(request.query["subarray"], wf["metadata"]["name"])
+            info.append(res)
+
+    buf = html_page("stop", subarray=request.query["subarray"], data=info)
+    return web.Response(body=buf, content_type="text/html")
+
+
+async def status_handle(request):
+    """
+    ---
+    description: status a subarray
+    parameters:
+       - in: query
+         name: subarray
+         schema:
+           type: integer
+         required: true
+         description: The subarray ID.
+    responses:
+        "200":
+            $ref: '#/components/responses/Reply200Ack'
+        "405":
+            $ref: '#/components/responses/HTTPMethodNotAllowed'
+        "421":
+            $ref: '#/components/responses/HTTPMisdirectedRequest'
+        "422":
+            $ref: '#/components/responses/HTTPUnprocessableEntity'
+    """
+    data = await subarray_status(request.query["subarray"])
+    buf = html_page("status", request.query["subarray"], data=data)
+    return web.Response(body=buf, content_type="text/html")
+
+
+async def map_page(request):
+    with open("index.html") as fh:
+        text = fh.read()
+    return web.Response(text=text.strip(), content_type="text/html")
+
+
+async def status_runner(app):
+    while True:
+        # app["status_obj"].check()
+        await asyncio.sleep(1)
+
+
+async def start_background_tasks(app):
+    app["status_runner"] = asyncio.Task(status_runner(app))
+
+
+app = web.Application()
+# app["status_obj"] = SdpPcStatus()
+app.on_startup.append(start_background_tasks)
+
+
+if __name__ == "__main__":
+    swagger = SwaggerDocs(
```

Comment:
We should read in a config file.
It will be important to get the namespace per subarray from config. Since multiple people will use the same cluster for testing I'd like it to get namespaces from config.

[default]
argo_url=blabla
argo_token_file=blabla
subarrays=4
serviceaccount=workflow

[subarray1]
k8s_namespace=sdparray1

...

Reply:
OR maybe the config should be yaml then it will display nice if we make it a configmap object in Kubernetes. 

---
argo_url: blabla
argo_token_file: blabla
serviceaccount: workflow
subarrays:
    - subarray1:
         k8s_namespace: sdparray1
    - subarray2:
         k8s_namespace: sdparray2

Reply:
This will be followed up in https://github.com/ska-sa/katsdpk8spoc/issues/4

Reply:
This will be followed up in https://github.com/ska-sa/katsdpk8spoc/issues/4